src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/VariantsSparkSink.java

```diff
@@ -178,7 +178,7 @@ private static void writeVariantsSingle(
             final VCFHeader header, final boolean writeGvcf, final List<Integer> gqPartitions, final int defaultPloidy, final int numReducers) throws IOException {
 
         final Configuration conf = ctx.hadoopConfiguration();
-        if (outputFile.endsWith(BGZFCodec.DEFAULT_EXTENSION)) {
+        if (outputFile.endsWith(BGZFCodec.DEFAULT_EXTENSION) || outputFile.endsWith(".gz")) {
```

Comment:
It's crazy that we don't have a consistent way we check this.  The best list seems to be in `AbstractFeatureReader.hasBlockCompressedExtensions` but there are a whole bunch of places that only accept .gz, or some combination of gz other extensions.